packages/wallet/src/TxBuilder/types.ts

```diff
@@ -0,0 +1,125 @@
+import { Cardano } from '@cardano-sdk/core';
+import { CustomError } from 'ts-custom-error';
+import { InputSelectionError } from '@cardano-sdk/cip2';
+import { OutputValidation } from '../types';
+import { SignTransactionOptions } from '../KeyManagement';
+
+export type DeepPartial<T> = T extends object
+  ? {
+      [P in keyof T]?: DeepPartial<T[P]>;
+    }
+  : T;
+
+export enum TxOutputFailure {
+  minimumCoin = 'Minimum coin not met',
+  tokenBundleSizeExceedsLimit = 'Token Bundle Exceeds Limit',
+  missingRequiredFields = 'Mandatory fields address or coin are missing'
+}
+
+export class OutputValidationMissingRequired extends CustomError {
```

Comment:
All other error types in the SDK have an `Error` suffix. Let's follow that for consistency.

```suggestion
export class OutputValidationMissingRequiredError extends CustomError {
```